pycytominer/cyto_utils/cell_locations.py

```diff
@@ -0,0 +1,401 @@
+                "Image and Nuclei tables are not present in the single_cell file"
+            )
+
+        # Verify that the required columns are present in the single_cell file
+
+        c.execute("PRAGMA table_info(Nuclei);")
+
+        nuclei_columns = c.fetchall()
+
+        nuclei_columns = [x[1] for x in nuclei_columns]
+
+        if not (
+            self.image_column in nuclei_columns
+            and self.object_column in nuclei_columns
+            and self.cell_x_loc in nuclei_columns
+            and self.cell_y_loc in nuclei_columns
+        ):
+            raise ValueError(
+                f"Required columns are not present in the Nuclei table in the SQLite file"
+            )
+
+        c.execute("PRAGMA table_info(Image);")
+
+        image_columns = c.fetchall()
+
+        image_columns = [x[1] for x in image_columns]
+
+        if not (
+            self.image_column in image_columns
+            and all(elem in image_columns for elem in self.image_index)
+        ):
+            raise ValueError(
+                f"Required columns are not present in the Image table in the SQLite file"
+            )
+
+        image_index_str = ", ".join(self.image_index)
+
+        # merge the Image and Nuclei tables in SQL
+
+        merge_query = f"""
+        SELECT Nuclei.{self.image_column},Nuclei.{self.object_column},Nuclei.{self.cell_x_loc},Nuclei.{self.cell_y_loc},Image.{image_index_str}
+        FROM Nuclei
+        INNER JOIN Image
+        ON Nuclei.{self.image_column} = Image.{self.image_column};
+        """
+
+        merged_df = pd.read_sql_query(merge_query, conn)
+
+        conn.close()
+
+        # if the single_cell file was downloaded from S3, delete the temporary file
+        if "temp_single_cell_input" in locals():
+            pathlib.Path(temp_single_cell_input).unlink()
+
+        # Cast the cell location columns to float
+        merged_df[self.cell_x_loc] = merged_df[self.cell_x_loc].astype(float)
+        merged_df[self.cell_y_loc] = merged_df[self.cell_y_loc].astype(float)
+
+        # Cast the object column to int
+        merged_df[self.object_column] = merged_df[self.object_column].astype(int)
+
+        # Cast the image index columns to str
+        for col in self.image_index:
+            merged_df[col] = merged_df[col].astype(str)
+
+        merged_df = self._convert_to_per_row_dict(merged_df)
+
+        return merged_df
+
+    def add_cell_location(self):
+        """Add the X,Y locations of all cells to the metadata file in the corresponding row, packed into a single column.
+        Optionally, save the augmented metadata file as a Parquet file.
+
+        Returns
+        -------
+        Pandas DataFrame
+            Either a data frame or the path to a Parquet file with the X,Y locations of all cells packed into a single column
+        """
+
+        # If self.augmented_metadata_output is not None and it is a str and the file already exists, there is nothing to do
```

Comment:
For now, I've added this comment:

```py
            # TODO: Consider doing a quick difference check should the file already exist.
            # For example, if the file already exists and it's different than what could be possibly incoming, should the user know?
            # This will involve performing all the steps below and then doing a check to see if the file is different, so this is a bit of a pain.

```